### ISSUE TITLE
[Bug]: --background not working

### ISSUE THREAD

**Issue by LDprg** [nextcloud/desktop#5916]
### ⚠️ Before submitting, please verify the following: ⚠️

- [X] This is a **bug**, not a question or a configuration issue.
- [X] This issue is **not** already reported on Github (I've searched it).
- [X] Nextcloud Server and Desktop Client are **up to date**. See [Server Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) and [Desktop Releases](https://nextcloud.com/install/#install-clients) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/)

### Bug description

I am autostarting nextcloud on linux with --background, but it still opens the window and does not minimize to tray.

### Steps to reproduce

1. Start nextcloud client with --background

### Expected behavior

Well, that the app does not show up on start or when using --background.

### Which files are affected by this bug

?

### Operating system

Linux

### Which version of the operating system you are running.

Arch with Gnome as DE

### Package

Distro package manager

### Nextcloud Server version

-

### Nextcloud Desktop Client version

2:3.9.0-2

### Is this bug present after an update or on a fresh install?

Fresh desktop client install

### Are you using the Nextcloud Server Encryption module?

Encryption is Enabled

### Are you using an external user-backend?

- [ ] Default internal user-backend
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Nextcloud Server logs

```shell
?
```

### Additional info

[background_bug.zip](https://github.com/nextcloud/desktop/files/12138935/background_bug.zip)

**Comment by LDprg** [comment]
After a couple of restarts (3-5) it worked again, but I let the issue open just in case.

**Comment by LDprg** [comment]
Strangely the problem accrued since today again.

**Comment by siSSSou** [comment]
The option "--background" sometimes works, but most of the time main window is shown at startup.
It's strange this bug is still alive...

**Comment by LDprg** [comment]
> The option "--background" sometimes works, but most of the time main window is shown at startup. It's strange this bug is still alive...

Yeah the issue seems to accrue only sometimes and also depending on the system. On my other Laptop it works as expected.expected

**Comment by WhoAmI0501** [comment]
I am also facing this behavior on my Ubuntu machines with Gnome... quite annoying.

**Comment by joshtrichards** [comment]
Related: #5382

**Comment by WhoAmI0501** [comment]
> Related: #5382

In my case, i want to add the information, that the window stays open until I explicitly close it via the task bar. Its not just popping up for a short moment.

**Comment by GUSC68** [comment]
Same here since updating the appimage to 3.14.0 on Debian/XFCE.
Start in the middle of the screen after the 1st login and disappear only after a clic on the icon in the notification area.
Worked like a charm before the update :/

**Comment by Dravere** [comment]
This is happening for me constantly now. The window always appears on startup. To reproduce I can restart the PC or exit the NextCloud client and start it with the `--background` parameter from a terminal and it will always show up with the main window. It's really annoying.

System is XUbuntu 24.04.1, NextCloud desktop client is at 3.14.1

**Comment by Jerome-Herbinet** [comment]
Same with Linux Mint XFCE 22 and 3.14.1 client.

**Comment by rotanid** [comment]
Also happens on Arch Linux for a longer while now, currently with XFCE 4.20 and Nextcloud Client 3.15.3

**Comment by ordilibre** [comment]
Also happens on Linux Mint Xia with XFCE 4.18.4 and Nextcloud Client 3.16.2daily (Linux)

**Comment by csit07** [comment]
This bug also occurs under Windows 10 since client version 3.14.

**Comment by bpmartin20** [comment]
This is intermittent, but  happens *almost* every time. Running under Ubuntu 24.04LTS and XFCE 4.18 using nextcloud-client 3.16. I tried to run strace on it (ran a short script that exec's strace launching nextcloud-client), and it *never* happens. Either the involvement of the wrapper script or strace prevented the problem. I'm testing now with just the wrapper script and no strace. If that still fails, it will suggest to me that this might be a timing issue that the presence of strace prevents. I'll report back.

**Comment by bpmartin20** [comment]
Reporting back ... 
The autolaunch file ("~/.config/autostart/nextcloud.desktop") would normally contain the following line, which causes the problem we're discussing -- the nextcloud client often usually ignores the --background option and appears in the foreground.
`Exec=/usr/bin/nextcloud --background`
If I change it to:
`Exec=/usr/bin/strace -o /tmp/nextcloud.trace -f /usr/bin/nextcloud --background`
it still fails intermittently, but frequently, just the same as before.
If I change it to:
`Exec=/home/bpmartin20/bin/launch-nextcloud`
and /home/bpmartin20/bin/launch-nextcloud contains:
`exec /usr/bin/nextcloud --background`
it still fails. But if I change /home/bpmartin20/bin/launch-nextcloud to contain:
`exec strace -o /tmp/nextcloud.trace -f /usr/bin/nextcloud --background`
it runs for a week without an incident. Running a trace on nextcloud all the time is a thorough waste of system resources, but maybe it suggests areas to investigate.

**Comment by siSSSou** [comment]
I have just installed an nvidia card in my computer and since then the same behavior occurs.
With intel gpu driver I never encounterd that problem, and with proprietary nvidia drivers it occurs at every startup !
(Debian 13 xfce)

**Comment by kolAflash** [comment]
There seem to be issues with multiple KDE / Qt apps, not showing systray icons on XFCE.  
- https://bugs.kde.org/show_bug.cgi?id=353062 _Konversation: Freeze/failure to start when "show tray icon" is enabled_
- flameshot-org: https://github.com/flameshot-org/flameshot/issues/4138#issuecomment-3805829657

It may be related to the need to have some _"StatusNotifier service"_ registered on D-Bus.  
https://forum.xfce.org/viewtopic.php?id=15497 _Void Linux, using xfce4-panel_

**Comment by bpmartin20** [comment]
Update:

Strike what I posted below. It's still happening. :frowning_face:

---

Google Gemini has apparently solved this for me. The solution was to turn off the client's native auto-start feature (if on) and to create or modify the ~/.config/autostart/nextcloud.desktop file to exec bash and run a 5 second sleep before launching nextcloud. Here's mine:

```
[Desktop Entry]
Version=1.0
Name=nextcloud
GenericName=nextcloud
Comment=nextcloud
Exec=/bin/bash -c "sleep 5 && /usr/bin/nextcloud --background"
Terminal=false
Type=Application
Categories=Office
StartupNotify=true
Hidden=false
X-GNOME-Autostart-enabled=true
RunHook=0

NoDisplay=true
# XFCE specific hint to ensure it loads after the window manager is stable
X-XFCE-Autostart-Override=true
```

This file may not be optimal. I've tried a lot of solutions, so there may be unnecessary elements in it.

Gemini's explanation is:
> On XFCE, launching the Nextcloud client simultaneously with the desktop can trigger a timing issue,  which bypasses the --background flag. The 5-second delay gives the system tray and session time to load first, ensuring the client launches directly into the background.

That explanation also explains why my earlier inclusion of an strace made the problem go away most of the time. The strace addition slowed down the start-up enough to be similar to the 5 second sleep.

Hope this helps, and ideally I hope the developers can incorporate something to eliminate the need for this hack.

**Comment by bpmartin20** [comment]
OK, here's a crude workaround that seems to be working for me. I don't like it much, but it's better that what I had.

In my ~/.local/autostart/nextcloud.desktop file, I replaced the Exec= statement wiith:

```
Exec=bash -c "for i in {1..20}; do if wmctrl -c 'Nextcloud' 2>/dev/null || xdotool windowminimize \$(xdotool search --onlyvisible --class 'nextcloud' | head -n 1) 2>/dev/null; then break; fi; sleep 0.5; done & /usr/bin/nextcloud --background"
```

On login, the nextcloud client appears briefly, but is killed and replaced with the background process. It required that I install the xdotool and wmctrl packages (one of which was already present).

Essentially, this automates the process of killing the unminimized window so I don't have to do it manually. Ugly, but better than nothing.
